qiskit/providers/ibmq/runtime/runtime_program.py

```diff
@@ -255,3 +257,78 @@ class ProgramResult(NamedTuple):
     name: str
     description: str
     type: str
+
+
+class ParameterNamespace(SimpleNamespace):
+    """ An abstraction for SimpleNamespace that offers param validation.
+
+    Note: Unused/Unnecessary params does not cause error (still validates) """
+
+    def __init__(self, params: List[ProgramParameter]):
+        """ParameterNamespace constructor.
+
+            Args:
+                params (List[ProgramParameter]): The program's input parameters
+        """
+        super().__init__()
+        # Allow access to the raw program parameters list
+        self.__metadata = params
+        # For localized logic, create store of parameters in dictionary
+        self.__program_params: dict = {}
+
+        for param in params:
+            # (1) Add parameters to a dict by name
+            setattr(self, param.name, None)
+            # (2) Store the program params for validation
+            self.__program_params[param.name] = param
+
+    @property
+    def metadata(self) -> List[ProgramParameter]:
+        """Returns the parameter metadata"""
+        return self.__metadata
+
+    def validate(self) -> None:
+        """Validates the user's usage of the program's inputs
+
+        Raises:
+            IBMQInputValueError if validation fails
+        """
+
+        # Iterate through the user's stored inputs
+        for param_name, program_param in self.__program_params.items():
+            # Set invariants: User-specified parameter value (value) and whether it's required (req)
+            value = getattr(self, param_name, None)
+            # Check there exists a program parameter of that name.
+            if value is None and program_param.required:
+                raise IBMQInputValueError('Param (%s) missing required value!' % param_name)
+
+            value_type = type(value).__name__
+            # Check program param of that name is of same type
+            if value is not None and value_type != program_param.type:
+                raise IBMQInputValueError('Parameter (%s) is incorrect type! (%s != %s)' % \
+                                          (param_name, value_type, program_param.type))
```

Comment:
I thought you removed type validation after our last conversation?

Reply:
Fixed! I forgot to push this 😅